third-party/chpl-venv/venv-use-sys-python.py

```diff
@@ -4,32 +4,97 @@
+
+This fixes up the chpl-venv installation in a Cray Chapel module that was built using a non-standard
+Python installation (ie, the Python used was not installed as "/usr/bin/python*").
+
+1. Remove the python executables from the virtualenv.
+   This effectively forces the virtualenv scripts to use the system python.
+   Without this, "start_test" will not work if the Cray Chapel module is installed on a different
+   machine (ie, if no Python is installed in the same location as when the module was built).
+
+2. Rewrite shebang lines (in scripts) in the virtualenv that point to the non-standard Python exe.
+   This change is probably not needed for most situations.
+
+As noted, these fixes are only applicable when building a Cray Chapel module under certain conditions.
+This has been tested only for the cray-xc platform, where the Cray Chapel module was:
+ - built on a cray-xc CLE 5 (system Python version 2.6, non-standard user-installed Python 2.7), and
+ - installed on a cray-xc CLE 6 (system Python version 2.7).
+"""
+
+
+def rm_python_bins(bindir):
+
+    """ This removes any file objects named "python", "python2", etc found in bindir.
+
+    bindir is the abs path to ./third-party/chpl-venv/$(platform)/$(python_version)/chpl-virtualenv/bin
+    """
+
+    if os.path.isdir(bindir) and not os.path.islink(bindir):
+        for f in os.listdir(bindir):
+            abs_f = os.path.join(bindir, f)
+            # abs_f is the abs path to some file system object under bindir
+            if f in [ 'python', 'python2', 'python2.6', 'python2.7' ]:
+                print('Removing {0}'.format(abs_f))
+                os.remove(abs_f)
+    else:
+        print('error, rm_python_bins(): {0} not found'.format(bindir))
+        sys.exit(2)
+
+
+def rw_shebangs(root):
+
+    """ This finds and rewrites shebang lines that point to a non-standard Python executable.
+    The new shebang points to the equivalent Python executable from a "standard" system Python installation
+      (ie, "/usr/bin/python*").
+
+    root is the abs path to $(chpl_home)/third-party/chpl-venv/$(platform)/$(python_version)
+    """
+
+    if os.path.isdir(root) and not os.path.islink(root):
+        for f in os.listdir(root):
+            rw = False
+
+            abs_f = os.path.join(root, f)
+            # abs_f is the abs path to some file system object under root
+            if os.path.isfile(abs_f) and not os.path.islink(abs_f):
+                # abs_f is a file
+                with open(abs_f) as fin:
+                    # read file abs_f and examine the first line
+                    lines = fin.readlines()
+                    if len(lines) > 0:
+                        line = lines[0]
+                        if re.match(r'#!/.*/bin/python', line):
+                            # first line is a shebang that points to python
+                            if not re.match(r'#!/usr/bin/python', line):
```

Comment:
This is fine, since it's consistent with previous regex match, but for funsies- this is clearer and faster:

```python

if '#!/usr/bin/python' not in line:

```